### ISSUE TITLE
Modelica.Mechanics.MultiBody.Examples.Loops.Fourbar2  is overconstrained at initialisation

### ISSUE THREAD

**Issue by modelica-trac-importer** [modelica/Modelica#945]
**Reported by jens&#x2e;frenkel on 27 Dec 2012 15:29 UTC**
The model Modelica.Mechanics.MultiBody.Examples.Loops.Fourbar2 is overconstrained at initialisation because j1.phi is fixed=true and j2.s is also fixed = true. However the systems has only one degree of freedom and consequently j1.s depents on j1.phi. Hence the startvalue of j2.s should not be fixed = true but used as guess value do avoid initialization in the second possible position.

----

Migrated-From: https://trac.modelica.org/Modelica/ticket/945

**Comment by modelica-trac-importer** [comment]
**Comment by otter on 1 Jan 2013 21:16 UTC**
Replying to [ticket:945 jens.frenkel@…]:

> The model Modelica.Mechanics.MultiBody.Examples.Loops.Fourbar2 is overconstrained at initialisation because j1.phi is fixed=true and j2.s is also fixed = true. However the systems has only one degree of freedom and consequently j1.s depents on j1.phi. Hence the startvalue of j2.s should not be fixed = true but used as guess value do avoid initialization in the second possible position.

This analysis is not correct. The reason is that parameter "computeRodLength" of component "universalSpherical" is set to true. This means that the length of the universalSpherical rod is not given (= the distance between the universal and the spherical joint) but is computed during initialization. For this, one additional initialization condition has to be defined. In the example the rod length is computed so that j2.s = -0.2. To summarize: The initialization is well defined in the example.

**Comment by modelica-trac-importer** [comment]
**Comment by christian&#46;schubert on 2 Jan 2013 09:08 UTC**
The reasoning is clear for me from a mechanical point of view but I am a bit puzzled by the Modelica-Implementation. Looking at Modelica.Mechanics.MultiBody.Joints.UniversalSpherical one sees that rodLength is a  parameters with fixed=false and a binding equation
```
final parameter SI.Distance rodLength(fixed=not computeRodLength)=
    Modelica.Math.Vectors.length(rRod_ia) 
```
As far as I understand, to get a balanced initial system I would have to discard the binding equation as soon as computeRodLength=true and therefore rodLength is fixed=false. 
However, the specification 3.3, Chapter 8.6, p. 91 says:
```
In the case a parameter has both a binding equation and fixed = false 
a diagnostics is recommended, but the parameter should be solved from 
the binding equation.
```
which I read as always keep the binding equation of an unfixed parameter.

It seems that Dymola always discards the binding equation for parameters with fixed=false as
```
model UnfixedParameter
  parameter Real A(fixed = false) = 2;
end UnfixedParameter;
```
gives the error 
```
The initial conditions are not fully specified.
Unfortunately, they cannot be completed by assuming start values for states.
There are 1 scalar conditions missing.
Cannot decide the following variables uniquely
A
```

Now I am wondering what the correct behavior is?

(It seems that this might be related to #923)

**Comment by modelica-trac-importer** [comment]
**Comment by otter on 2 Jan 2013 17:24 UTC**
Fixed in 6292fefb8c27c0a0d1726561462da1e51c21699a:

Replying to [comment:2 christian.schubert@…]:

> The reasoning is clear for me from a mechanical point of view but I am a bit puzzled by the Modelica-Implementation. Looking at Modelica.Mechanics.[wiki:MultiBody].Joints.[wiki:UniversalSpherical] one sees that rodLength is a  parameters with fixed=false and a binding equation
> ```
> final parameter SI.Distance rodLength(fixed=not computeRodLength)=
> Modelica.Math.Vectors.length(rRod_ia) 
> ```
> As far as I understand, to get a balanced initial system I would have to discard the binding equation as soon as computeRodLength=true and therefore rodLength is fixed=false.

You are right, the declaration of parameter rodLength in model UniversalSpherical is not correct. Changed this to:

```
final parameter SI.Distance rodLength(fixed=false,
                               start=Modelica.Math.Vectors.length(rRod_ia))
initial equation
  if not computeRodLength then
    rodLength = Modelica.Math.Vectors.length(rRod_ia);
  end if;

```

**Comment by modelica-trac-importer** [comment]
**Modified by otter on 2 Jan 2013 17:24 UTC**

**Comment by modelica-trac-importer** [comment]
**Comment by christian&#46;schubert on 2 Jan 2013 21:54 UTC**
I've just seen, that a similar problem exists for 
```
Modelica.Mechanics.MultiBody.Joints.Assemblies.JointRRR
```
with the parameter n_b defined as
```
final parameter Real n_b[3](each final unit="1",fixed=false) = {0,0,1};
```
having the initial equation
```
n_b = Frames.resolve2(frame_b.R, Frames.resolve1(frame_a.R, n_a));
```

**Comment by modelica-trac-importer** [comment]
**Comment by pharman on 3 Jan 2013 15:12 UTC**
This is indeed the same issue as #923.

I've written a test to scan the MSL for parameters with `fixed=false` and a binding equation and `n_b` in `Modelica.Mechanics.MultiBody.Joints.Assemblies.JointRRR` does appear to be the last remaining instance of this.

**Comment by modelica-trac-importer** [comment]
**Comment by otter on 5 Jan 2013 17:48 UTC**
Fixed in f15953962ebf7a4e7fbbf7e4436d9cae1c63a0db
